java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -264,10 +467,65 @@ public ColumnVector copyToDevice() {
     }
   }
 
+  private HostMemoryBuffer getData() {
+    HostMemoryBuffer ret = null;
+    if (children.size() > 0)
+      return children.get(0).getData();
+    else return offHeap.data;
+  }
+
+  private List<HostMemoryBuffer> getValids() {
+    List<HostMemoryBuffer> ret = new ArrayList<>();
+    ret.add(offHeap.valid);
+    for (NestedHostColumnVector child : children) {
+      ret.addAll(child.getValids());
+    }
+    return ret;
+  }
+
+  private List<HostMemoryBuffer> getOffsets() {
+    List<HostMemoryBuffer> ret = new ArrayList<>();
+    ret.add(offHeap.offsets);
+    for (NestedHostColumnVector child : children) {
+      ret.addAll(child.getOffsets());
+    }
+    return ret;
+  }
+
+  private List<Long> getRows() {
+    List<Long> ret = new ArrayList<>();
+    ret.add(rows);
+    for (NestedHostColumnVector child : children) {
+      ret.addAll(child.getRows());
+    }
+    return ret;
+  }
+
+  private List<DType> getTypes() {
+    List<DType> ret = new ArrayList<>();
+    ret.add(type);
+    for (NestedHostColumnVector child : children) {
+      ret.addAll(child.getTypes());
+    }
+    return ret;
+  }
   /////////////////////////////////////////////////////////////////////////////
   // DATA ACCESS
   /////////////////////////////////////////////////////////////////////////////
 
+  public List getList(long rowIndex) throws Exception {
```

Comment:
If we are going to expose this we need lots and lots of comments to say that this is very slow and should never be used in production.